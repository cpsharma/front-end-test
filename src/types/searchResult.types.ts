export interface OptionsType {
	label: string;
	value: string;
};

export interface RatingAndFacilitiesType extends OptionsType {
	isChecked: boolean;
	name: string;
};

export interface PricePerPersonType extends RatingAndFacilitiesType {
	min: string;
	max: string;
}