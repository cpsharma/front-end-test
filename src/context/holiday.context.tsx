import { h, JSX, createContext } from "preact";
import { prependOnceListener } from "process";
import { Holiday } from "../types/booking";

const HolidayContext = createContext<Holiday[]>([]);
export { HolidayContext };
