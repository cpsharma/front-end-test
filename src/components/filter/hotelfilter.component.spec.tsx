import { h } from "preact";
import { mount, configure, shallow, ShallowWrapper } from "enzyme";
import Adapter from "enzyme-adapter-preact-pure";
import { HolidayContext } from "../../context/holiday.context";
import HotelFilterComponent from "./hotelfilter.component";
import holidaysMock from "../../__mock__/holiday.json";
import { BookingResponse } from "../../types/booking";

configure({ adapter: new Adapter() });

describe("HotelFilter Component", () => {
  const holidaysMockObj: BookingResponse = JSON.parse(
    JSON.stringify(holidaysMock)
  );
  let hotel_filter_component;

  beforeEach(() => {
    hotel_filter_component = mount(
      <HolidayContext.Provider value={holidaysMockObj.holidays}>
        <div>
          <HotelFilterComponent />
        </div>
      </HolidayContext.Provider>
    );
  });
  afterEach(() => {
    hotel_filter_component.unmount();
  });
  it("renders properly", async () => {
    expect(hotel_filter_component).toMatchSnapshot();
  });

  it("Renders Child components `StarRatingFilterComponent, SliderComponent, FacilityFilterComponent, HotelCardComponent`", async () => {
    expect(
      hotel_filter_component.find("ul#start-rating-filter").length
    ).toEqual(1);
  });
  it("Renders Child components `StarRatingFilterComponent, SliderComponent, FacilityFilterComponent, HotelCardComponent`", async () => {
    expect(
      hotel_filter_component.find("input#price-range-filter").length
    ).toEqual(1);
  });
  it("Renders Child components `StarRatingFilterComponent, SliderComponent, FacilityFilterComponent, HotelCardComponent`", async () => {
    expect(hotel_filter_component.find("ul#facility-filter").length).toEqual(1);
  });
  it("Renders Child components `StarRatingFilterComponent, SliderComponent, FacilityFilterComponent, HotelCardComponent`", async () => {
    expect(hotel_filter_component.find("ul#facility-filter").length).toEqual(1);
  });
});
