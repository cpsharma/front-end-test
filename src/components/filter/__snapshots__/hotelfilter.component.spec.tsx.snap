// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelFilter Component renders properly 1`] = `ReactWrapper {}`;
